pkg/state/partials.go

```
@@ -0,0 +1,207 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+
+	memdb "github.com/hashicorp/go-memdb"
+	"github.com/kong/go-database-reconciler/pkg/utils"
+)
+
+const (
+	partialTableName = "partial"
+)
+
+var partialTableSchema = &memdb.TableSchema{
+	Name: partialTableName,
+	Indexes: map[string]*memdb.IndexSchema{
+		"id": {
+			Name:    "id",
+			Unique:  true,
+			Indexer: &memdb.StringFieldIndex{Field: "ID"},
+		},
+		"name": {
+			Name:         "name",
+			Unique:       true,
+			Indexer:      &memdb.StringFieldIndex{Field: "Name"},
+			AllowMissing: true,
+		},
+		all: allIndex,
+	},
+}
+
+// PartialsCollection stores and indexes Kong Partials.
+type PartialsCollection collection
+
+func getPartial(txn *memdb.Txn, IDs ...string) (*Partial, error) {
+	for _, id := range IDs {
+		res, err := multiIndexLookupUsingTxn(txn, partialTableName,
+			[]string{"name", "id"}, id)
+		if errors.Is(err, ErrNotFound) {
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		partial, ok := res.(*Partial)
+		if !ok {
+			panic(unexpectedType)
```

Ditto: shouldn't we return an error instead?